File: inventario-frontend/models/Producto.ts
import { Categoria } from "./Categoria";

export interface Producto {
  id: number;
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  categoria_id: number;
  categoria?: Categoria;
  created_at: string;
  updated_at: string;
}

export interface ProductoRequest {
  nombre: string;
  descripcion: string;
  precio: number;
  stock: number;
  categoria_id: number;
}

File: inventario-backend/internal/models/producto.go
package models

import "time"

type Producto struct {
	ID          int       `json:"id"`
	Nombre      string    `json:"nombre"`
	Descripcion string    `json:"descripcion"`
	Precio      float64   `json:"precio"`
	Stock       int       `json:"stock"`
	CategoriaID int       `json:"categoria_id"`
	Categoria   *Categoria `json:"categoria,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ProductoRequest struct {
	Nombre      string  `json:"nombre"`
	Descripcion string  `json:"descripcion"`
	Precio      float64 `json:"precio"`
	Stock       int     `json:"stock"`
	CategoriaID int     `json:"categoria_id"`
}

